### DISCUSSION TITLE
Way to set `initialPlacement` with

### DISCUSSION THREAD

**Discussion by konsalex** [floating-ui/floating-ui#2197] [General]
Is there any way to set `initialPlacement` while using the `useFloating` hook?

**Comment by atomiks** [comment]
What do you mean by `initialPlacement` in this context?

**Reply by konsalex** [reply]
This: https://floating-ui.com/docs/middleware#initialplacement

**Reply by konsalex** [reply]
Basically if we allow 4 positions, to have an initial preferred placement to work with

**Reply by atomiks** [reply]
The regular `placement` option is that value. The `placement` returned from `useFloating` is the final one.

**Reply by konsalex** [reply]
To give a more concrete example: 

I have this `useFloating` usage, and while the placement is `top` and allowed placements are `top` and `bottom`, in the initial calculation `bottom` is calculated and was curious if I could make it prefer `top` with not much of a boilerplate logic

```tsx
  const { context, x, y, reference, floating, strategy, placement } =
    useFloating({
      open,
      onOpenChange: setOpen,
      whileElementsMounted: autoUpdate,
      placement: 'top',
      middleware: [
        offset(10),
        autoPlacement({
          allowedPlacements,
        }),
      ],
    });
```

https://user-images.githubusercontent.com/12672541/222166418-ccc57631-faec-4e1e-bf30-dca72abb06f6.mp4

**Reply by atomiks** [reply]
It looks like you want to use `flip()` not `autoPlacement()`

**Reply by konsalex** [reply]
Thanks @atomiks will give it a try, though `flip` was only for one axis
